### PR TITLE
[T3A3][T16-A1]Lai Zhiyuan

### PR DESCRIPTION
**Pull request by zyjames**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/AddressBlock.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    /**
+     * Default constructor of AddressBlock
+     */
+    public AddressBlock() {}
+    
+    /**
+     * Constructor to initialise the block of the address
+     * @param block
+     */
+    public AddressBlock(String block) {
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/AddressBlock.java]: may be it would help to do the validation of individual blocks?

**Review by okkhoy — COMMENTED** [review]
this is a decent attempt at the exercise. please take a look at the specific comments and suggestions for improvement.

**Comment by okkhoy** [comment]
@zyjames  some comments added, please take a look. please close the PR once you have noted the suggestions.

File: .settings/org.eclipse.jdt.ui.prefs [hunk 2]
```diff
@@ -0,0 +1,4 @@
+cleanup_settings_version=2
+eclipse.preferences.version=1
```
[the remark this answers is not in the document]
Comment (zyjames) [reply .settings/org.eclipse.jdt.ui.prefs:2]: Will look up on how to put this file using git ignore! Thanks for the guidance!(:

File: doc/UserGuide.md [hunk 3]
```diff
@@ -34,7 +34,7 @@ Format: `help`
  
 ### Adding a person: `add`
 Adds a person to the address book<br>
-Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/ADDRESS [t/TAG]...` 
+Format: `add NAME [p]p/PHONE_NUMBER [p]e/EMAIL [p]a/BLOCK, STREET, UNIT, POSTAL_CODE [t/TAG]...`
```
[the remark this answers is not in the document]
Comment (zyjames) [reply doc/UserGuide.md:37]: I think perhaps I should put [p]a/BLOCK [COMMA] STREET [COMMA] UNIT [COMMA] POSTAL_CODE?

File: src/seedu/addressbook/commands/CommandResult.java [hunk 4]
```diff
@@ -11,7 +11,7 @@
 public class CommandResult {
 
     /** The feedback message to be shown to the user. Contains a description of the execution result */
-    public final String feedbackToUser;
+    private final String feedbackToUser;
```
[the remark this answers is not in the document]
Comment (zyjames) [reply src/seedu/addressbook/commands/CommandResult.java:14]: So sorry about it, cause i merge that 2 branch, thus it appeared in this activity.

File: src/seedu/addressbook/data/person/Address.java [hunk 5]
```diff
@@ -8,12 +10,19 @@
...
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in this format: a/BLOCK, STREET, UNIT, POSTAL_CODE";
+    public static final String ADDRESS_VALIDATION_REGEX = "[\\d\\w]{1,}[,\\s].{1,}[,\\s].{1,}[,\\s][\\d]+";
+    public static final String ADDRESS_DELIMITER = ",";
 
-    public final String value;
+    //public final String value;
```
[the remark this answers is not in the document]
Comment (zyjames) [reply src/seedu/addressbook/data/person/Address.java:18]: Noted (:

File: src/seedu/addressbook/data/person/Address.java [hunk 6]
```diff
@@ -25,32 +34,62 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        } else {
+            Scanner addressScanner = new Scanner(trimmedAddress).useDelimiter(ADDRESS_DELIMITER);
+            setAddress(addressScanner.next(), addressScanner.next(), addressScanner.next(), addressScanner.next());
```
[the remark this answers is not in the document]
Comment (zyjames) [reply src/seedu/addressbook/data/person/Address.java:39]: If one doesn't add the comma, the validation would have failed, thus the code will not execute this line. (:

File: src/seedu/addressbook/data/person/AddressBlock.java (hunk 1 above)
Comment (zyjames) [reply src/seedu/addressbook/data/person/AddressBlock.java:16]: How would you recommend to validate the block other than checking that it is not empty?

**Pull request closed without merging**